packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -493,14 +472,51 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		localId: string,
 		blob: ArrayBufferLike,
 	): Promise<ICreateBlobResponse | void> {
-		return PerformanceEvent.timedExecAsync(
+		return runWithRetry(
+			async () => {
+				try {
+					return await this.getStorage().createBlob(blob);
+				} catch (error) {
+					const entry = this.pendingBlobs.get(localId);
+					assert(
+						!!entry,
+						0x387 /* Must have pending blob entry for blob which failed to upload */,
+					);
+					if (entry.opsent && !canRetryOnError(error)) {
+						if (isFluidError(error)) {
+							error.addTelemetryProperties({ forceRetry: true });
+							throw error;
+						} else {
+							throw wrapError(error, () =>
+								createGenericNetworkError(
```

Comment:
I don't think createGenericNetworkError is really appropriate here, as we don't know anything about the inner error, so we don't know if it is network or not. i think just creating a new logging error would be enough, and wrap error will take most the properties from the inner error.

Reply:
the biggest problem is that this approach overwrites the errorType, using logging error will preserve it.